lib/sensu/client/process.rb

```diff
@@ -209,11 +209,11 @@ def run_check_extension(check)
       # @param check [Hash]
       def process_check_request(check)
         @logger.debug("processing check", :check => check)
+        if @settings.check_exists?(check[:name])
+          check.merge!(@settings[:checks][check[:name]])
+        end
         if check.has_key?(:command)
-          if @settings.check_exists?(check[:name])
-            check.merge!(@settings[:checks][check[:name]])
-            execute_check_command(check)
-          elsif @safe_mode
+          if !@settings.check_exists?(check[:name]) && @safe_mode
```

Comment:
let's flip it around, for readability: `@safe_mode && !@settings.check_exists?(check[:name])`

:+1: